lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -604,9 +608,32 @@ newDBLayer trace defaultFieldValues mDatabaseFile = do
         , putDelegationCertificate = \(PrimaryKey wid) cert sl -> ExceptT $ do
             selectWallet wid >>= \case
                 Nothing -> pure $ Left $ ErrNoSuchWallet wid
-                Just _  -> pure <$> repsert
-                    (DelegationCertificateKey wid sl)
-                    (DelegationCertificate wid sl (W.dlgCertPoolId cert))
+                Just _  -> case cert of
+                    W.CertDelegateNone _ -> do
+                        repsert
+                            (DelegationCertificateKey wid sl)
+                            (DelegationCertificate wid sl Nothing)
+                        pure <$> repsert
+                            (StakeKeyCertificateKey wid sl)
+                            (StakeKeyCertificate wid sl False)
+                    W.CertDelegateFull _ pool ->
+                        pure <$> repsert
+                            (DelegationCertificateKey wid sl)
+                            (DelegationCertificate wid sl (Just pool))
+                    W.CertRegisterKey _ ->
+                        pure <$> repsert
+                            (StakeKeyCertificateKey wid sl)
+                            (StakeKeyCertificate wid sl True)
+
+        , readIsStakeKeyRegistered = \(PrimaryKey wid) -> ExceptT $ do
+            val <- fmap entityVal <$> selectFirst
+                    [StakeKeyCertWalletId ==. wid]
+                    [Desc StakeKeyCertSlot]
+
+            return $ case val of
+                Nothing -> Left $ ErrNoSuchWallet wid
```

Comment:
Hmmm. No. There could be a wallet but no key. This should return negatively about the existing of a key, but not return an error. To know whether there's a wallet, this should use the same pattern as for other functions:

```hs
            selectWallet wid >>= \case
                Nothing -> pure $ Left $ ErrNoSuchWallet wid
                Just{} -> {- ... do actual stuff ... -}
```